services/dkg/pedersen/controller/action.go

```diff
@@ -48,38 +55,16 @@ func (a *initAction) Execute(ctx node.Context) error {
 
 	ctx.Injector.Inject(actor)
 	dela.Logger.Info().Msg("DKG has been initialized successfully")
-	return nil
-}
-
-// linkToElectionAction is an action to link a pre-existing DKG service to an election
-//
-// - implements node.ActionTemplate
-type linkToElectionAction struct {
-}
 
-// Execute implements node.ActionTemplate. It creates an actor from
-// the dkgPedersen instance
-func (a *linkToElectionAction) Execute(ctx node.Context) error {
 	var dkgMap kv.Bucket
 	err := ctx.Injector.Resolve(&dkgMap)
 	if err != nil {
 		return xerrors.Errorf("failed to resolve dkgMap: %v", err)
 	}
 
-	var dkg dkg.DKG
-	// TODO: This doesn't seem to work
-	err := ctx.Injector.Resolve(&dkg)
-	if err != nil {
-		return xerrors.Errorf("failed to resolve dkg: %v", err)
-	}
-
-	electionIDBuf, err := hex.DecodeString(ctx.Flags.String("electionID"))
-	if err != nil {
-		return xerrors.Errorf("failed to decode electionID: %v", err)
-	}
-
-	// TODO: dkg is not a byte array
-	dkgMap.Set(dkg, electionIDBuf)
+	// TODO: How to use the fact that actor has factory and context attributes?
+	actorBuf := serde.Serialize(actor.context)
```

Comment:
Type `Actor` has a `context` attribute, yet I get the compile-error: `actor.context undefined (type "github.com/dedis/d-voting/services/dkg".Actor has no field or method context)`. How to remedy this?

Reply:
How do I use the `serde` module to make `actor` into a byte array?